de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/EditorManager.java

```diff
@@ -852,21 +861,46 @@ public void closeEditor(final SPath path) {
   public void adjustViewport(
       @NotNull final SPath path, final LineRange range, final TextSelection selection) {
 
-    executeInUIThreadSynchronous(
-        () -> {
-          Editor editor = editorPool.getEditor(path);
+    Set<String> visibleFilePaths = new HashSet<>();
 
-          if (editor == null) {
-            LOG.warn(
-                "Failed to adjust viewport for "
-                    + path
-                    + " as it is not known to the editor pool.");
+    for (VirtualFile virtualFile : projectAPI.getSelectedFiles()) {
+      visibleFilePaths.add(virtualFile.getPath());
+    }
 
-            return;
-          }
+    VirtualFile passedFile = VirtualFileConverter.convertToVirtualFile(path);
 
-          localEditorManipulator.adjustViewport(editor, range, selection);
-        });
+    if (passedFile == null) {
+      LOG.warn(
+          "Ignoring request to adjust viewport as no valid VirtualFile could be found for "
+              + path
+              + " - given range: "
+              + range
+              + ", given selection: "
+              + selection);
+
+      return;
+    }
+
+    Editor editor = editorPool.getEditor(path);
+
+    if (!visibleFilePaths.contains(passedFile.getPath())) {
+      localEditorStatusChangeHandler.queueViewPortChange(
+          passedFile.getPath(), editor, range, selection);
+
+      return;
+    }
+
+    if (editor == null) {
+      LOG.warn(
+          "Failed to adjust viewport for "
+              + path
+              + " as it is not known to the editor pool even though it is currently open");
+
+      return;
+    }
+
+    ApplicationManager.getApplication()
```

Comment:
Either use `executeInUIThreadSynchronous` or `ApplicationManager...` directly and remove wrapper.